ionics_fits/models/utils.py

```diff
@@ -47,15 +49,17 @@ class PeriodicModelParameter(ModelParameter):
     scale_func: TSCALE_FUN
     period: float = 1
     offset: float = 0
-    lower_bound: float = dataclasses.field(init=False)
-    upper_bound: float = dataclasses.field(init=False)
+    lower_bound: float = dataclasses.field(init=False, default=-np.inf)
+    upper_bound: float = dataclasses.field(init=False, default=+np.inf)
```

Comment:
Would it make sense to expose these as `@property`s instead? This way we can just do something like this, and don't need the `__setattr__` override.

```python
@property
def lower_bound(self):
  return -np.inf

@property
def lower_bound(self):
  return +np.inf
```

Reply:
Ahh, of course. Sorry, hadn't thought about the fact that we're inheriting/overriding these properties from the superclass. 

It actually took me a moment to figure out how this works — the `init=False` causes the field to not be set in the constructor at all (I had assumed it still set it, just to the default), instead attribute lookup falls back to the default value.

> What do you think the right overall design is here?

Yeah, it's tricky. Some of this is personal preference, and I'm also not familiar with all the nitty-gritty of ionics_fits, so I don't think some of my initial thoughts make sense.

I think my inclination is let's leave it as-is for now (so keeping the use of `__setattr__`). It's not especially pretty, but it does actually work at least!

Reply:
yeah, I don't love this part of the design, but I don't see an obviously better approach. Happy to merge for now since this PR doesn't make the situation worse, and return to it later when someone has time and motivation